src/packages/@highlighter/node/HighlighterNode.ts

```diff
@@ -0,0 +1,65 @@
+import randomcolor from 'randomcolor';
+import { HighlighterCore } from '../core';
+
+export type HighlighterNodeRenderer = (keyword: string) => HTMLElement;
+
+export interface HighlighterNodeOptions {
+    /** Function is called on render founded keyword */
+    highlightRenderer: HighlighterNodeRenderer;
+    /** Instance of HighlighterCore library */
+    highlighterCore: HighlighterCore;
+}
+
+/**
+ * This number allows retrieve more different color
+ *
+ * Need for internal implementation 'randomcolor' library
+ * when we use 'seed' option for sequence numbers make its more different
+ */
+const COLOR_MULTIPLIER = 10;
+
+export class HighlighterNode {
+    private highlightRenderer: HighlighterNodeRenderer;
+
+    private highlighterCore: HighlighterCore;
+
+    /**
+     * Generate random colors
+     */
+    static generateColor(index: number) {
+        return randomcolor({
+            luminosity: 'light',
+            seed: index !== null ? index * COLOR_MULTIPLIER : null,
+        });
+    }
+
+    static applyStylesForNode(node: HTMLElement, styles: CSSStyleDeclaration) {
+        Object.keys(styles).forEach((property) => {
+            node.style[property] = styles[property]; // eslint-disable-line no-param-reassign
+        });
+    }
+
+    constructor({ highlightRenderer, highlighterCore }: HighlighterNodeOptions) {
+        this.highlightRenderer = highlightRenderer;
+        this.highlighterCore = highlighterCore;
+    }
+
+    public find(node: Node, onMatch: (node: HTMLElement) => void) {
+        if (!node.textContent || !this.highlighterCore) return;
```

Comment:
Can we replace this condition block with the one below so it would minimize the number of returns and to avoid using of negative `if` condition
```
if (node.textContent || this.highlighterCore) {
   /** the rest code of the function below **/
}
```

Reply:
yes, I replaced it.